src/cascadia/TerminalApp/AppLogic.cpp

```diff
@@ -345,6 +345,62 @@ namespace winrt::TerminalApp::implementation
             TraceLoggingBool(_settings.GlobalSettings().ShowTabsInTitlebar(), "TabsInTitlebar"),
             TraceLoggingKeyword(MICROSOFT_KEYWORD_MEASURES),
             TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage));
+
+        _onCreateResearch();
+    }
+
+    void AppLogic::_onCreateResearch()
+    {
+        // ----------------------------- RE: Themes ----------------------------
+        const auto numThemes = _settings.GlobalSettings().Themes().Size();
+        const auto themeInUse = _settings.GlobalSettings().CurrentTheme();
+        const auto usingDefaultTheme = themeInUse.Name() == L"light" ||
+                                       themeInUse.Name() == L"dark" ||
+                                       themeInUse.Name() == L"system";
+
+        TraceLoggingWrite(
+            g_hTerminalAppProvider,
+            "ThemesInUse",
+            TraceLoggingDescription("Data about the themes in use"),
+            TraceLoggingBool(usingDefaultTheme, "UsingDefaultTheme"),
+            TraceLoggingInt32(numThemes, "NumThemes"),
+            TraceLoggingKeyword(MICROSOFT_KEYWORD_MEASURES),
+            TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage));
+
+        // --------------------------- RE: sendInput ---------------------------
+
+        auto totalSendInput = 0;
+        const auto& allActions = _settings.GlobalSettings().ActionMap().AvailableActions();
+        for (const auto&& [name, actionAndArgs] : allActions)
+        {
+            if (actionAndArgs.Action() == ShortcutAction::SendInput)
+                totalSendInput++;
+        }
+
+        TraceLoggingWrite(
+            g_hTerminalAppProvider,
+            "SendInputUsage",
+            TraceLoggingDescription("Data about usage of sendInput"),
+            TraceLoggingInt32(totalSendInput, "NumSendInputActions"),
+            TraceLoggingKeyword(MICROSOFT_KEYWORD_MEASURES),
+            TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage));

```

Comment:
Can you do something more like this...


```
auto getTotalSendInputCount = [&](){
    // calculate how many Send Input actions there are
}

...

TraceLoggingInt32(getTotalSendInputCount(), "NumSendInputActions"),
...
```

`TraceLogging` is intentionally very lightweight. It only evaluates its arguments if tracing is enabled. Very, very explicitly.

It compiles to code like this.

```
if (_globalTraceLoggingHookEnabled_g_hTerminalAppProvider) {
   // all the TraceLoggingWrite code
}

```

If we don't guard evaluations safely, we effectively waste a bunch of CPU time for an event that isn't being listened to.